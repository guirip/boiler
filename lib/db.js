import client from 'nawr/client'
import { v4 as uuid } from 'uuid'
import bcrypt from 'bcrypt'
const SALT_ROUNDS = 8

export const createUser = async ({ email, password }) => {
  const hash = await bcrypt.hash(password, SALT_ROUNDS)

  const newUser = { email, password: hash, id: uuid() }
  await client.query(
    `INSERT INTO users (email,password,id) VALUES(:email,:password, uuid(:id))`,
    newUser
  )

  return newUser
}

export const getUserByEmail = async email => {
  const { records } = await client.query(
    `select * from users where email = :email`,
    {
      email
    }
  )

  return records[0]
}
