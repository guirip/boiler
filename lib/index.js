export function runMiddleware(req, res, fn) {
  return new Promise((resolve, reject) => {
    fn(req, res, result => {
      if (result instanceof Error) {
        return reject(result)
      }

      return resolve(result)
    })
  })
}

export const redirect = path => {
  return ({ res, req }) => {
    res.setHeader('location', path)
    res.statusCode = 302
    res.end()
  }
}
