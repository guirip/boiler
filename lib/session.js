import { withIronSession, applySession } from 'next-iron-session'

export const options = {
  password:
    process.env.SECRET_COOKIE_PASSWORD || 'your-thirty-two-character-secret',
  cookieName: 'auth-cookie',
  cookieOptions: {
    // the next line allows to use the session in non-https environements like
    // Next.js dev mode (http://localhost:3000)
    secure: process.env.NODE_ENV === 'production'
  }
}

export default function withSession(handler) {
  return withIronSession(handler, options)
}

export function apply(req, res) {
  return applySession(req, res, options)
}

export function login(req, user) {
  req.session.set('user', user)
  return req.session.save()
}
