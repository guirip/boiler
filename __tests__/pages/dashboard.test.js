import React from 'react'
import renderer from 'react-test-renderer'
import Dash, { handler as getServerSideProps } from '../../pages/dashboard'
import { ServerResponse, IncomingMessage, request, Headers } from 'http'
import { apply, login } from '../../lib/session'

it('Redirects to login page with unauthed user', async () => {
  const req = new IncomingMessage({
    method: 'GET'
  })
  const res = new ServerResponse(req)
  await apply(req, res)

  const props = await getServerSideProps({
    req,
    res
  })

  expect(res.statusCode).toBe(302)
})

it('Renders correctly with logged in user', async () => {
  // TODO create user in db first

  const user = {
    email: 'x@x.com',
    password: '123'
  }

  const req = new IncomingMessage({
    method: 'GET'
  })
  const res = new ServerResponse(req)

  await apply(req, res)
  await login(req, user)

  const { props } = await getServerSideProps({
    req,
    res
  })

  expect(props.user).toEqual(user)
  const tree = renderer.create(<Dash {...props} />).toJSON()
  expect(tree).toMatchSnapshot()
})
