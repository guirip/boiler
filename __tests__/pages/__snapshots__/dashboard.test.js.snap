// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly with logged in user 1`] = `
<div>
  <h3>
    hi ðŸ‘‹ 
    x@x.com
  </h3>
  <p>
    Here is a list of all users stored in the db
  </p>
  <ul />
</div>
`;
